### DISCUSSION TITLE
nodebuilder/p2p: Consider redesign of network determination/configuration

### DISCUSSION THREAD

**Discussion by Wondertan** [celestiaorg/celestia-node#3038] [Ideas]
~Currently, we only have vars defined in ‘params’~

Consider redesigning the network determination/configuration functionality since it is split between parsing env for custom network vs using `--p2p.network` to choose from a select amount of (valid) networks and it is confusing especially for those who rely on this functionality for testing / spinning up custom networks.

**Comment by Bidon15** [comment]
Grooming 16/11/2022:

Moving to post-mainnet
